### PR TITLE
Update custom classnames with  prefix

### PR DESCRIPTION
**Pull request by namanaman**
## Overview
- We added in custom button classes to enable styling of NJWDS buttons in ways that USWDS did not support
- To make this clearer that it is NJ-specific, I've added an `nj-` prefix to classnames

## Test plan
1. Ran `npm run build-uswds` to build the latest stylesheet
2. Ran `npm run build-docs` to build updated docs (with new classnames used)
3. Ran `npm run start` to run the docs locally
4. Compare local docs with production docs (https://newjersey.github.io/njwds/index.html) to see there's no regressions

### CODE REVIEW

File: src/components/02-buttons/buttons.config.yml
Comment (ezhangy) [inline src/components/02-buttons/buttons.config.yml:1]: [dust] Out of scope for this PR, but wondering whether we should look into why the icon leading and icon only buttons are offset. I inspected the CSS briefly but couldn't see an obvious reason why that should be the case
Fractal docs for primary light button variant

**Review by ezhangy — APPROVED** [review]
Looks good! Thanks again for making this update 🙏

One out-of-scope dust and one suggestion for renaming the NJ custom classnames to a different convention (which you should please feel to take or leave!)

File: src/sass/_uswds-theme-custom-styles.scss
[the remark this answers is not in the document]
Comment (namanaman) [reply src/sass/_uswds-theme-custom-styles.scss:1]: Oh good idea! I like the convention, lines up with USWDS and it's easier to read as well. I can change them, it should be quick

File: src/components/02-buttons/buttons.config.yml
Comment (namanaman) [reply src/components/02-buttons/buttons.config.yml:1]: Yea ugh I spent some time trying to understand the difference but I had no idea. If this ends up affecting code outside of the demo site, then will prioritize further

**Pull request merged**